interpreter/k8s/src/main/scala/io/buoyant/transformer/k8s/DaemonSetTransformer.scala

```diff
@@ -0,0 +1,73 @@
+package io.buoyant.transformer.k8s
+
+import com.twitter.finagle.Name.Bound
+import com.twitter.finagle.{Addr, Name, Address, NameTree}
+import com.twitter.util.{Var, Activity}
+import io.buoyant.namer.{DelegateTree, DelegatingNameTreeTransformer}
+
+/**
+ * The DaemonSetTransformer maps each address in the NameTree to a member of
+ * a given daemon set that is on the same /24 subnet.  Since each k8s node
+ * is its own /24 subnet, the result is that each address is mapped to the
+ * member of the daemon set that is running on the same node.  This can be used
+ * to redirect traffic to a reverse-proxy that runs as a daemon set.
+ */
+class DaemonSetTransformer(daemonSet: Activity[NameTree[Bound]]) extends DelegatingNameTreeTransformer {
+
+  override protected def transformDelegate(tree: DelegateTree[Bound]): Activity[DelegateTree[Bound]] = {
+    daemonSet.map { daemonSet =>
+      val daemons = flatten(daemonSet.eval.toSet.flatten)
+      tree.map(mapBound(_, daemons))
+    }
+  }
+
+  override protected def transform(tree: NameTree[Bound]): Activity[NameTree[Bound]] = {
+    daemonSet.map { daemonSet =>
+      val daemons = flatten(daemonSet.eval.toSet.flatten)
+      tree.map(mapBound(_, daemons))
+    }
+  }
+
+  /** Smoosh together all of the bound addresses into a single Var */
+  private[this] def flatten(bounds: Set[Name.Bound]): Var[Addr] = {
+    Var.collect(bounds.map(_.addr)).map { addrs =>
+      val collectedAddresses = addrs.flatMap {
+        case Addr.Bound(addresses, _) => addresses
+        case _ => Set.empty[Address]
+      }
```

Comment:
tioli: could this just be:
```scala
addrs.collect { case Addr.Bound(sas, _) => sas }.flatten
```

(i don't think i have a preference, though this one seems more concise)